src/items.rs

```diff
@@ -1958,20 +1969,29 @@ fn rewrite_explicit_self(
             )?;
 
             Some(format!(
-                "{}self: {}",
+                "{}{}self: {}",
+                param_attrs,
                 format_mutability(mutability),
                 type_str
             ))
         }
-        ast::SelfKind::Value(mutability) => Some(format!("{}self", format_mutability(mutability))),
+        ast::SelfKind::Value(mutability) => Some(format!(
+            "{}{}self",
+            param_attrs,
+            format_mutability(mutability)
+        )),
     }
 }
 
 pub(crate) fn span_lo_for_arg(arg: &ast::Param) -> BytePos {
-    if is_named_arg(arg) {
-        arg.pat.span.lo()
+    if arg.attrs.is_empty() {
```

Comment:
Nit: this is not related to this PR, but I will just mention it in case you find yourself interested. Please feel free to ignore this.

Recently libsyntax had renamed `Arg` to `Param`. I thought that I have renamed all variables whose name is `arg` and type is `Param` to `param` in the rustfmt codebase, but there are plenties of unfinished work like this :( It would be great if you could refactor these misuses.

Reply:
Definitely. I'll do that refactoring in a follow up PR if that sounds good to you?

Reply:
> Definitely. I'll do that refactoring in a follow up PR if that sounds good to you?


Awesome, sounds good to me!